### DISCUSSION TITLE
Should setting status on a Response parameter work?

### DISCUSSION THREAD

**Discussion by dave-shawley** [fastapi/fastapi#12247] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi
import pydantic


class Status(pydantic.BaseModel):
    status: str


app = fastapi.FastAPI()


@app.get('/status')
def get_status(*, response: fastapi.Response) -> Status:
    response.status = 304
    return Status(status='ok')
```

### Description

Retrieve `/status` from the server using `curl` or any other tool that shows the status code. I was expecting a `304 Not Modified` response and received a `200 OK`.
```
% curl -v http://127.0.0.1:8888/status | jq
*   Trying 127.0.0.1:8888...
* Connected to 127.0.0.1 (127.0.0.1) port 8888
> GET /status HTTP/1.1
> Host: 127.0.0.1:8888
> User-Agent: curl/7.86 (daveshawley)
> Accept: */*
>
< HTTP/1.1 200 OK
< date: Sun, 22 Sep 2024 14:35:24 GMT
< server: uvicorn
< content-length: 15
< content-type: application/json
<
{ [15 bytes data]
* Connection #0 to host 127.0.0.1 left intact
{
  "status": "ok"
}
```

After a little debugging (and then reading the documentation 😞), I switched from `response.status` to `response.status_code` and it works correctly.

> My question is whether setting `Response.status` should work as well.

I can put together a PR to implement this is it should work.

### Operating System

macOS

### Operating System Details

macOS 14.4, M2 MacBook Pro

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.12.4

**Comment by Kludex** [comment, marked as answer]
... No. `status` shouldn't work. The `status_code` should, and works, as you mentioned.

<img width="936" alt="Screenshot 2024-09-22 at 17 24 20" src="https://github.com/user-attachments/assets/440b7c77-1c1c-4519-9e93-89e994a930a5">

Also, the type checkers do identify that `response.status` is not valid.

**Reply by dave-shawley** [reply]
🤦 now I have to figure out why type errors are not showing in my PyCharm configuration. Thanks.
